### PR TITLE
vimPlugins.copilot-vim: init at 2021-12-07

### PR DESCRIPTION
**Pull request by malob**
###### Motivation for this change
I'd like to have access to this plugin in `nixpkgs`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vim-plugins/deprecated.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "compe-tmux": {
-        "date": "2021-12-07",
+        "date": "2021-12-10",
```
Comment (figsoda) [inline pkgs/misc/vim-plugins/deprecated.json:3]: i think the update script is doing something weird, can you drop this commit?

cc @teto if you want to take a look at the update script

Comment (malob) [reply pkgs/misc/vim-plugins/deprecated.json]: Done.

File: pkgs/misc/vim-plugins/generated.nix [hunk 2]
```diff
@@ -7260,7 +7272,7 @@ final: prev:
       owner = "flazz";
       repo = "vim-colorschemes";
       rev = "fd8f122cef604330c96a6a6e434682dbdfb878c9";
-      sha256 = "1cg8q7w0vgl73aw1b9zz0zh5vw5d2pm8pm54fhfzva4azg56f416";
+      sha256 = "0kpsf6j20fgblc8vhqn7ymr52v2d1h52vc7rbxmxfwdm80nvv3g5";
```
Comment (figsoda) [inline pkgs/misc/vim-plugins/generated.nix:7275]:
```suggestion
      sha256 = "1cg8q7w0vgl73aw1b9zz0zh5vw5d2pm8pm54fhfzva4azg56f416";
```

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 150143` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>69 packages built:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.ale</li>
    <li>vimPlugins.chadtree</li>
    <li>vimPlugins.cmp-buffer</li>
    <li>vimPlugins.cmp-tabnine</li>
    <li>vimPlugins.comment-nvim</li>
    <li>vimPlugins.conjure</li>
    <li>vimPlugins.copilot-vim</li>
    <li>vimPlugins.crates-nvim</li>
    <li>vimPlugins.diffview-nvim</li>
    <li>vimPlugins.doki-theme-vim</li>
    <li>vimPlugins.dracula-vim</li>
    <li>vimPlugins.falcon</li>
    <li>vimPlugins.friendly-snippets</li>
    <li>vimPlugins.gentoo-syntax</li>
    <li>vimPlugins.gitsigns-nvim</li>
    <li>vimPlugins.julia-vim</li>
    <li>vimPlugins.lean-nvim</li>
    <li>vimPlugins.lightspeed-nvim</li>
    <li>vimPlugins.lir-nvim</li>
    <li>vimPlugins.lsp-status-nvim</li>
    <li>vimPlugins.lsp_signature-nvim</li>
    <li>vimPlugins.lualine-nvim</li>
    <li>vimPlugins.luasnip</li>
    <li>vimPlugins.minimap-vim</li>
    <li>vimPlugins.neogit</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.neovim-ayu</li>
    <li>vimPlugins.nightfox-nvim</li>
    <li>vimPlugins.nui-nvim</li>
    <li>vimPlugins.null-ls-nvim</li>
    <li>vimPlugins.nvim-autopairs</li>
    <li>vimPlugins.nvim-bqf</li>
    <li>vimPlugins.nvim-dap</li>
    <li>vimPlugins.nvim-gdb</li>
    <li>vimPlugins.nvim-hlslens</li>
    <li>vimPlugins.nvim-jdtls</li>
    <li>vimPlugins.nvim-lspconfig</li>
    <li>vimPlugins.nvim-metals</li>
    <li>vimPlugins.nvim-spectre</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.plenary-nvim</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.rust-tools-nvim</li>
    <li>vimPlugins.symbols-outline-nvim</li>
    <li>vimPlugins.tagbar</li>
    <li>vimPlugins.telescope-cheat-nvim</li>
    <li>vimPlugins.telescope-frecency-nvim</li>
    <li>vimPlugins.telescope-fzf-native-nvim</li>
    <li>vimPlugins.telescope-fzf-writer-nvim</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.telescope-symbols-nvim</li>
    <li>vimPlugins.telescope-z-nvim</li>
    <li>vimPlugins.trouble-nvim</li>
    <li>vimPlugins.vim-airline</li>
    <li>vimPlugins.vim-devicons</li>
    <li>vimPlugins.vim-fireplace</li>
    <li>vimPlugins.vim-flagship</li>
    <li>vimPlugins.vim-fugitive</li>
    <li>vimPlugins.vim-git</li>
    <li>vimPlugins.vim-sleuth</li>
    <li>vimPlugins.vim-test</li>
    <li>vimPlugins.vim-togglelist</li>
    <li>vimPlugins.vim-wakatime</li>
    <li>vimPlugins.vimspector</li>
    <li>vimPlugins.vimtex</li>
    <li>vimPlugins.which-key-nvim</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**